arelle/plugin/validate/HMRC/ValidateHmrc.py

```diff
@@ -314,9 +330,96 @@ def _evaluateCode(self, code: str) -> CodeResult:
                     ))
         return self._setCode(code, CodeResult())
 
+    def _evaluateAuditFacts(self, prefix: str) -> CodeResult:
+        """
+        Logs an error when an audited report does not facts tagged with the concepts of "DateAuditorsReport",  "OpinionAuditorsOnEntity"
+        as well as either "NameIndividualAuditor" or the combination of "NameSeniorStatutoryAuditor" and "NameEntityAuditors".
+        :param prefix:
+        :return:
+
+        """
+        code = ''
+        if prefix == COMPANY:
+            code = CO_AUDIT
+        elif prefix == LLP:
+            code = LP_AUDIT
+        missingConcepts = []
+        if not self._getAndCheckValidFacts(CONCEPT_DATE_AUDITOR_REPORT):
+            missingConcepts.append(CONCEPT_DATE_AUDITOR_REPORT)
+        if not self._getAndCheckValidFacts(CONCEPT_OPINION_AUDITORS_ON_ENTITY):
+            missingConcepts.append(CONCEPT_OPINION_AUDITORS_ON_ENTITY)
+        if (not (self._getAndCheckValidFacts(CONCEPT_NAME_ENTITY_AUDITORS) and self._getAndCheckValidFacts(CONCEPT_NAME_SENIOR_STATUTORY_AUDITOR))
+                and not self._getAndCheckValidFacts(CONCEPT_NAME_INDIVIDUAL_AUDITOR)):
+            missingConcepts.append(CONCEPT_NAME_INDIVIDUAL_AUDITOR)
+            missingConcepts.append(CONCEPT_NAME_SENIOR_STATUTORY_AUDITOR)
+            missingConcepts.append(CONCEPT_NAME_ENTITY_AUDITORS)
+        if len(missingConcepts) > 0:
+            return self._setCode(
+                code,
+                CodeResult(
+                    success=False,
+                    conceptList=missingConcepts,
+                    message="An audited report must contain facts tagged with the concepts of DateAuditorsReport, OpinionAuditorsOnEntity"
+                            "as well as either NameIndividualAuditor or the combination of NameSeniorStatutoryAuditor and NameEntityAuditors"
+                            "There are no facts tagged with the concepts: %(conceptLocalName)s"
```

Comment:
It doesn't look like `conceptList` gets used. I'm guessing you want it here instead of `conceptLocalName`.